### ISSUE TITLE
Can't attach a file in application documents directory

### ISSUE THREAD

**Issue by 12people** [taljacobson/flutter_mailer#10]
When I try to attach a file in my application documents directory (path "/data/user/0/app.emparent.android/app_flutter/file.db"), I get the error `PlatformException (PlatformException(error, Failed to find configured root that contains /data/data/app.emparent.android/app_flutter/file.db, null))`. (Note the slight difference in the interpreted path.)

I'm 100% sure that the path is correct (it's the same path that I use for saving and reading the database, and that works). I'm using getApplicationDocumentsDirectory() from path_provider to get the directory.

**Comment by 12people** [comment]
Seems like flutter_email_sender, a similar library, also has this problem: https://github.com/sidlatau/flutter_email_sender/issues/10

**Comment by taljacobson** [comment]
sorry for the late response.
it might have something to do with android permissions for writing and reading from external.

personally i create a temp file in my projects.

**Comment by 12people** [comment]
I do so now to.
It'd be nice to have a fix down the line, though.

**Comment by CloudyPadmal** [comment]
Yep, both the libraries got this issue. Workaround for me is to copy the file in app directory and set that path as attachment. 

```js

getTemporaryDirectory().then((tempDir) {
    File("${directory-in-app-path}.zip")
        .copySync("${tempDir.path}/temp.zip");
    final MailOptions mailOptions = MailOptions(
        body: 'BODY',
        subject: 'SUBJECT',
        recipients: ['example@example.com'],
        isHTML: true,
        attachments: [
            "${tempDir.path}/item_finder_backup.zip",
        ],
    );
    FlutterMailer.send(mailOptions).then((v) {
        // TODO: Delete the temp file
    }).catchError((errE) {
        print(errE);
    });
});
```

In native Android, probably related to external path configurations. Refer to this [StackOverflow Q](https://stackoverflow.com/questions/44184653/illegalargumentexception-failed-to-find-configured-root-that-contains-filepath).

**Comment by Aluzja** [comment]
Running into a similar issue but the error I'm seeing is:

`Exception has occurred.
PlatformException (PlatformException(error, Attempt to invoke virtual method 'android.content.res.XmlResourceParser android.content.pm.ProviderInfo.loadXmlMetaData(android.content.pm.PackageManager, java.lang.String)' on a null object reference, null))`

I attempted to copy the file I'm trying to send into the cache and use that:

`var tempDirectory = await getTemporaryDirectory();
    var file = File(logProvider.databasePath).copySync('${tempDirectory.path}/log_db.db');

    final MailOptions mailOptions = MailOptions(
      subject: 'SUBJECT',
      body: 'BODY,
      recipients: ['something@something.com'],
      isHTML: false,
      attachments: [file.path],
    );

    await FlutterMailer.send(mailOptions);`

**Comment by langaro** [comment]
I had the same issue so I used:
`    final directory = await getTemporaryDirectory();`

**Comment by ricardogobbo** [comment]
I use External Directory from path_provider pub
```
final directory = await getExternalStorageDirectory();
file = new Io.File('${directory.path}/image.png');
```

**Comment by Damandroid** [comment]
Facing the same issue with getApplicationsDocumentDirectory(). Tried with getExternalStorageDirectory() but it only works for android and I do not want to store my generated file externally. Any suggestions which could work on both platforms?

**Comment by fralways** [comment]
I cannot get this to work on --release build, tried all kind of directories. iOS and Android both fail to send attachment.

**Comment by frankvollebregt** [comment]
Hi! I was trying around, and while copying the file to a temporary directory first seems to work for me, I would very much like to be able to add attachments from my application files directly.

During my search, I stumbled upon a comment on [an issue in React Native](https://github.com/react-native-share/react-native-share/issues/218#issuecomment-396460146). It proposes adding the root of the application directory to the paths in which the `FileProvider` used can look. Since this plugin uses a similar system, I was able to locally add `<root-path name="root" path="." />` to the `res/xml/flutter_mailer_paths.xml` file, after which I could add files from the application folder on both Android devices I tested with.
